union-type.js

```diff
@@ -127,4 +127,22 @@ function Type(desc) {
 
 Type.check = true;
 
+Type.ListOf = function (T) {
+  var List = Type({List:[Array]});
+  var innerType = Type({T: [T]}).T;
+  var validate = List.case({
+    List: function (array) {
+      try{
+        for(var n in array) {
```

Comment:
This is not guaranteed to iterate in the correct order on ES5 and lower, and it can accidentally iterate over non-numeric properties when prototypes are monkey patched. This would be much safer as:

```
for (var i = 0; i < array.length; i++)
```